crates/rustc_codegen_spirv/src/codegen_cx/constant.rs

```diff
@@ -374,23 +386,88 @@ impl<'tcx> CodegenCx<'tcx> {
-                    64 => self.constant_f64(DUMMY_SP, f64::from_bits(v as u64)),
-                    other => self
+            SpirvType::Bool
+            | SpirvType::Integer(..)
+            | SpirvType::Float(_)
+            | SpirvType::Pointer { .. } => {
+                let (primitive, size) = match ty_concrete {
+                    SpirvType::Bool => (
+                        Primitive::Int(Integer::fit_unsigned(0), false),
+                        SpirvType::Bool.sizeof(self).unwrap(),
+                    ),
+                    SpirvType::Integer(int_size, int_signedness) => {
+                        let integer = match int_size {
+                            8 => Integer::I8,
+                            16 => Integer::I16,
+                            32 => Integer::I32,
+                            64 => Integer::I64,
+                            128 => Integer::I128,
+                            other => {
+                                self.tcx
+                                    .sess
+                                    .fatal(&format!("invalid size for integer: {}", other));
+                            }
+                        };
+                        (
+                            Primitive::Int(integer, int_signedness),
+                            SpirvType::Integer(int_size, int_signedness)
+                                .sizeof(self)
+                                .unwrap(),
+                        )
+                    }
+                    SpirvType::Float(float_size) => {
+                        let float_size_primitive = match float_size {
+                            32 => Primitive::F32,
+                            64 => Primitive::F64,
+                            other => {
+                                self.tcx
+                                    .sess
+                                    .fatal(&format!("invalid size for float: {}", other));
+                            }
+                        };
+                        (
+                            float_size_primitive,
+                            SpirvType::Float(float_size).sizeof(self).unwrap(),
+                        )
+                    }
+                    SpirvType::Pointer {
+                        storage_class,
+                        pointee,
+                    } => (
+                        Primitive::Pointer,
+                        SpirvType::Pointer {
+                            storage_class,
+                            pointee,
+                        }
+                        .sizeof(self)
+                        .unwrap(),
+                    ),
+                    unsupported_spirv_type => bug!(
+                        "invalid spirv type internal to create_alloc_const2: {:?}",
+                        unsupported_spirv_type
+                    ),
+                };
+                let scalar = match alloc
+                    .read_scalar(self, Pointer::new(AllocId(0), *offset), size)
```

Comment:
maybe a comment would be good pointing out how `AllocId` isn't used and therefore doesn't need to be filled in, it's pretty confusing as-is.